colabs/wandb_registry/zoo_wandb.ipynb

```diff
@@ -0,0 +1,984 @@
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "6432936b-31bf-477b-a4aa-a3a85345e5aa",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Decsribe how we split the training dataset for future reference, reproducibility.\n",
+    "config = {\n",
+    "    \"random_state\" : 42,\n",
+    "    \"test_size\" : 0.25,\n",
+    "    \"shuffle\" : True\n",
+    "}"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "5243bcd0-c3bc-4958-bb62-4ac129cbaed2",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Split dataset into training and test set\n",
+    "X_train, X_test, y_train, y_test = train_test_split(\n",
+    "    dataset,labels, \n",
+    "    random_state=config[\"random_state\"],\n",
+    "    test_size=config[\"test_size\"], \n",
+    "    shuffle=config[\"shuffle\"]\n",
+    ")\n",
+    "\n",
+    "# Save the files locally\n",
+    "torch.save(X_train, \"zoo_dataset_X_train.pt\")\n",
+    "torch.save(y_train, \"zoo_labels_y_train.pt\")\n",
+    "\n",
+    "torch.save(X_test, \"zoo_dataset_X_test.pt\")\n",
+    "torch.save(y_test, \"zoo_labels_y_test.pt\")"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "028d0992-15a9-4687-8c3b-4218bf70adb9",
+   "metadata": {},
+   "source": [
+    "Next, let's publish this dataset into a different collection within the Dataset registry called \"zoo-dataset-tensors-split\":"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "a64bacb4-5ca7-4a74-abc8-e890c655fe17",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "run = wandb.init(\n",
+    "    entity=TEAM_ENTITY,\n",
+    "    project=PROJECT,\n",
+    "    job_type=\"publish_split_dataset\", \n",
+    "    config=config\n",
+    ")\n",
+    "\n",
+    "# Let's add a description to let others know which file to use in future experiments\n",
+    "artifact = wandb.Artifact(\n",
+    "    name=\"split_zoo_dataset\",\n",
+    "    type=\"dataset\", \n",
+    "    description=\"Artifact contains `zoo_dataset` split into 4 datasets. \\\n",
+    "                For training, use `zoo_dataset_X_train` and `zoo_labels_y_train`. \\\n",
+    "                For testing, use `zoo_dataset_X_test` and `zoo_labels_y_test`.\"\n",
+    ")\n",
+    "\n",
+    "artifact.add_file(local_path=\"zoo_dataset_X_train.pt\", name=\"zoo_dataset_X_train\")\n",
+    "artifact.add_file(local_path=\"zoo_labels_y_train.pt\", name=\"zoo_labels_y_train\")\n",
+    "artifact.add_file(local_path=\"zoo_dataset_X_test.pt\", name=\"zoo_dataset_X_test\")\n",
+    "artifact.add_file(local_path=\"zoo_labels_y_test.pt\", name=\"zoo_labels_y_test\")\n",
+    "\n",
+    "REGISTRY_NAME = \"Dataset\"\n",
+    "COLLECTION_NAME = \"zoo-dataset-tensors-split\"\n",
+    "target_dataset_path=f\"{ORG_NAME}/wandb-registry-{REGISTRY_NAME}/{COLLECTION_NAME}\"\n",
+    "\n",
+    "run.link_artifact(artifact=artifact, target_path=target_dataset_path)\n",
+    "\n",
+    "run.finish()"
```

Comment:
Potentially a good idea to use a context manager here too, it helps visualize how little code they actually have to add in order to log to W&B.